Development/src/pages/devices/ChannelMappingMatrix.js

```diff
@@ -0,0 +1,514 @@
+                    >
+                        <div>{input_item.properties.name}</div>
+                    </Tooltip>
+                </TableCell>
+                {!isRowExpanded(input_name) && (
+                    <FillEmptyCellsForCollapseRow
+                        outputs={outputs}
+                        isColExpanded={isColExpanded}
+                    />
+                )}
+            </TableRow>
+            {isRowExpanded(input_name) && (
+                <InputChannelsMappingRows
+                    input_item={input_item}
+                    input_nmos_name={input_name}
+                    outputs={outputs}
+                    isColExpanded={isColExpanded}
+                    mapping_disabeld={is_show}
+                    handleMap={handleMap}
+                    isMapped={isMapped}
+                />
+            )}
+        </>
+    ));
+};
+
+const SortByIOName = io_object => {
+    return Object.entries(io_object).sort((io_item1, io_item2) =>
+        io_item1[1].properties.name.localeCompare(io_item2[1].properties.name)
+    );
+};
+
+const ChannelMappingMatrix = ({ record, is_show, mapping, handleMap }) => {
+    const [expandedCols, setExpandedCols] = useState([]);
+    const [expandedRows, setExpandedRows] = useState([]);
+    const handleEpandRow = input_name => {
+        const currentExpandedRows = expandedRows;
+        const isRowExpanded = currentExpandedRows.includes(input_name);
+        const newExpandedRows = isRowExpanded
+            ? currentExpandedRows.filter(name => name !== input_name)
+            : currentExpandedRows.concat(input_name);
+        setExpandedRows(newExpandedRows);
+    };
+    const handleEpandCol = output_name => {
+        const currentExpandedCols = expandedCols;
+        const isColExpanded = currentExpandedCols.includes(output_name);
+        const newExpandedCols = isColExpanded
+            ? currentExpandedCols.filter(name => name !== output_name)
+            : currentExpandedCols.concat(output_name);
+        setExpandedCols(newExpandedCols);
+    };
+    const isRowExpanded = input_name => {
+        return expandedRows.includes(input_name);
+    };
+    const isColExpanded = output_name => {
+        return expandedCols.includes(output_name);
+    };
+    const isMapped = (
+        input_name,
+        output_name,
+        input_channel,
+        output_channel
+    ) => {
+        return (
+            input_name ===
+                get(mapping, `${output_name}.${output_channel}.input`) &&
+            String(input_channel) ===
+                String(
+                    get(
+                        mapping,
+                        `${output_name}.${output_channel}.channel_index`
+                    )
+                )
+        );
+    };
+    const io = get(record, '$io');
+    const sorted_outputs = SortByIOName(get(io, `outputs`));
+    const sorted_inputs = SortByIOName(get(io, `inputs`));
+    return (
+        <Table border={1}>
```

Comment:
What I'm seeing in my browser is a one pixel black border round the whole table, and one pixel black borders vertically between columns, but a one pixel pale grey border horizontally between rows. If we could all of those in the pale grey, I think that would match the look'n'feel of the list views, etc. better?